apps/project/serializers.py

```
@@ -73,21 +74,36 @@ class Meta:  # type: ignore[reportIncompatibleVariableOverride]
             "tutorial",
         )
 
-    def validate_status(self, new_status: Project.Status | int | None):
+    def validate_status(self, new_status: Project.Status | int) -> Project.Status:
         assert self.instance is not None
 
-        if new_status is None:
-            return None
         if isinstance(new_status, int):
             new_status = Project.Status(new_status)
 
-        if (self.instance.status_enum, new_status) not in VALID_PROJECT_STATUS_TRANSITIONS:
+        if self.instance.status != new_status and (self.instance.status, new_status) not in VALID_PROJECT_STATUS_TRANSITIONS:
             raise serializers.ValidationError(
                 gettext("Project status cannot be changed from %s to %s")
-                % (self.instance.status_enum.label, new_status.label),
+                % (
+                    self.instance.status_enum.label,
+                    new_status.label,
+                ),
             )
         return new_status
 
+    def validate_tutorial(self, tutorial: Tutorial | None) -> Tutorial | None:
+        assert self.instance is not None
+        current_tutorial = self.instance.tutorial
+
+        if current_tutorial and tutorial and tutorial != current_tutorial and tutorial.status == Tutorial.Status.ARCHIVED:
```

We don't need to check for `current_tutorial`
If there was no previous tutorial and we added an archived tutorial, we still need to throw an error.

---

See line 273 for reference.